resources/assets/js/editor.js

```diff
@@ -151,7 +153,7 @@ function addTab(file) {
     span.dataset.filename = file.filename;
     let text = document.createTextNode(file.filename);
     span.appendChild(text);
-    if (file.filename !== "main.py") {
+    if (file.filename !== "main.py" && clientIsAuthor) {
```

Comment:
Look for the readOnly variable in this file. We may need to move the declaration of readOnly but we should use this variable for this test if possible so we don't double up.